sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -176,17 +182,41 @@ function calculateQueryParameters(
   return result;
 }
 
+function simpleParseQueryParams(queryString: string): Array<[string, string]> {
+  if (!queryString || queryString[0] !== "?") {
+    return [];
+  }
+
+  // remove the leading ?
+  queryString = queryString.slice(1);
+
+  const pairs = queryString.split("&");
+
+  return pairs.map((pair) => {
+    const [name, value] = pair.split("=", 2);
```

Comment:
nit: this should be equivalent to 

```typescript
return pairs.map((pair) => pari.split("=", 2))
```

Reply:
I did it this way originally, but TS isn't smart enough to know the second parameter limits the size of the result and makes it into a 2 tuple, so it complains:

> Type 'string[][]' is not assignable to type '[string, string][]'.
> Type 'string[]' is missing the following properties from type '[string, string]': 0, 1